Bottom: 749879ecee71e1913edd443ed3f2949682c73dd1
Top:    90139e57fc2414b1c510b2456e31d337fbe308d5
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 14:54:50 +0800

autofs-5.0.7 - add initialization of bind_result in-do_sasl_bind().patch

There is an unlikley code path where bind_result could be used uninitialized
so initialize it so it isn't incorrectly used if it has rubish in it.


---

diff --git a/modules/cyrus-sasl.c b/modules/cyrus-sasl.c
index b456333..68f9242 100644
--- a/modules/cyrus-sasl.c
+++ b/modules/cyrus-sasl.c
@@ -210,7 +210,7 @@ int
 do_sasl_bind(unsigned logopt, LDAP *ld, sasl_conn_t *conn, const char **clientout,
 	     unsigned int *clientoutlen, const char *auth_mech, int sasl_result)
 {
-	int ret, msgid, bind_result;
+	int ret, msgid, bind_result = LDAP_OTHER;
 	struct berval client_cred, *server_cred, temp_cred;
 	LDAPMessage *results;
 	int have_data, expected_data;
